File: TP Evaluatorio_I_Aplicación_de_POO_en_otro_lenguaje._Hoy_tomamos_JAVA_y-PYTHON/Python/Interface/Interface.py
from abc import abstractmethod
from abc import ABCMeta


#Player Interface:
class Player():

    def __init__(self):
        pass
    def open(self,filePath):
        pass
    def play(self):
        pass
    def stop(self):
        pass
    def setVolume(self,value):
        pass


#PlayerPitchable Interface:
class PlayerPitchable():

    def setPitch(self,pitch):
        pass


#VLC concrete class
class VLC(Player,PlayerPitchable): 

    def __init__(self):

        print("The VLC constructor was invoked.")
        self.isOpen = False;
        self.isPlaying = False;
        self.setVolume(10);

    def __del__(self):

        print("The VLC destructor was invoked.")

    def open(self,filePath):

        #@ToDo: por ahora simulamos la apertura correcta
        self.isOpen = True;
        print("The audiofile: " + filePath + " is open.")

    def play(self):

        #@ToDo: por ahora simulamos la reproducción correcta
        if(self.isOpen) : self.isPlaying = True;
        print("The audiofile is playing.")

    def stop(self):

        #@ToDo: por ahora simulamos la detención correcta
        if(self.isPlaying) : self.isPlaying = False;
        print("The audiofile is stopped.")

    def setVolume(self, value):

        self.volume = value;
        print("The volume value is: " + str(self.volume))

    def setPitch(self, value):

        self.pitch = value;
        print("The pitch value is: " + str(self.pitch))

    #NO SE PUEDE CREAR UN OBJETO O INSTANCIA DE UNA INTERFAZ:
    #Player player;#NO!
vlcPlayer = VLC();
vlcPlayer.open("./resources/orchestral.ogg");
vlcPlayer.play();
vlcPlayer.setVolume(13);

File: TP Evaluatorio_I_Aplicación_de_POO_en_otro_lenguaje._Hoy_tomamos_JAVA_y-PYTHON/C++/Interface/Interface.cpp
#include <iostream>
#include <fstream>
using namespace std;

//Player Interface:
class Player
{
public:
    virtual void open(string filePath) = 0;
    virtual void play() = 0;
    virtual void stop() = 0;
    virtual void setVolume(float value) = 0;
};

//PlayerPitchable Interface:
class PlayerPitchable
{
public:
    virtual void setPitch(float pitch) = 0;
};

//VLC concrete class
class VLC : public Player, public PlayerPitchable
{
public:
VLC()
{
    cout << "The VLC constructor was invoked." << endl;
    isOpen = false;
    isPlaying = false;
    setVolume(10);
}
~VLC()
{
    cout << "The VLC destructor was invoked." << endl;
}
void open(string filePath)
{
    //@ToDo: por ahora simulamos la apertura correcta
    isOpen = true;
    cout << "The audiofile: " << filePath << " is open." << endl;
}
void play()
{
    //@ToDo: por ahora simulamos la reproducción correcta
    if(isOpen) isPlaying = true;
    cout << "The audiofile is playing." << endl;
}
void stop()
{
    //@ToDo: por ahora simulamos la detención correcta
    if(isPlaying) isPlaying = false;
    cout << "The audiofile is stopped." << endl;
}
void setVolume(float value)
{
    volume = value;
    cout << "The volume value is: " << volume << endl;
}
void setPitch(float value)
{
    pitch = value;
    cout << "The pitch value is: " << pitch << endl;
}

private:
    bool isOpen;
    bool isPlaying;
    float volume;
    float pitch;
};

int main()
{
    //NO SE PUEDE CREAR UN OBJETO O INSTANCIA DE UNA INTERFAZ:
    //Player player;//NO!
    VLC vlcPlayer;
    cout << endl;
    vlcPlayer.open("./resources/orchestral.ogg");
    vlcPlayer.play();
    vlcPlayer.setVolume(13);
    cout << endl;
return EXIT_SUCCESS;
}